### PR TITLE
handle addtestclient properly when not host

### PR DESCRIPTION
**Pull request by Lierrmm**
Fixes: #245

### CODE REVIEW

File: src/client/component/client_patches.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@ namespace client_patches
 	namespace
 	{
 		utils::hook::detour preload_map_hook;
+		utils::hook::detour sv_addtestclient_hook;
```
Comment (momo5502) [inline src/client/component/client_patches.cpp:16]: do you need a detour hook for that? would it be possible to i intercept all the calls to that function instead?

Comment (momo5502) [reply src/client/component/client_patches.cpp:16]: cause detours can potentially break if there are rip-relative instructions within the jump to the trampoline that gets inserted, those cannot properly be reconstructed by the reconstructed original stub

Comment (momo5502) [reply src/client/component/client_patches.cpp:16]: therefore using them with care and only where necessary is desired. or you make sure there are no rip-relative instructions, that also works.

Comment (Lierrmm) [reply src/client/component/client_patches.cpp:16]: I've just realized that the spawnBot command is defined in bots.cpp. 

Would it make more sense to add the logic there instead?

Comment (momo5502) [reply src/client/component/client_patches.cpp:16]: yes, definitely, just make sure your clientnum check is ignored for dedis

Comment (momo5502) [reply src/client/component/client_patches.cpp:16]: or better, extract it into a is_host function that always returns true on dedis

**Pull request closed without merging**